test/gateway_composer/accept_stake_request.js

```diff
@@ -12,22 +12,246 @@
+contract('GatewayComposer::acceptStakeRequest', async (accounts) => {
+    describe('Negative Tests', async () => {
+        const accountProvider = new AccountProvider(accounts);
+
+        it('Fails when requestHash is invalid.', async () => {
+            const {
+                gatewayComposer,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const {
+                facilitator,
+            } = await gatewayComposerUtils.setupGatewayPass(
+                accountProvider,
+            );
+
+            const r = web3.utils.soliditySha3('r');
+            const s = web3.utils.soliditySha3('s');
+            const v = 0;
+            const hashLock = web3.utils.soliditySha3('hl');
+
+            const invalidStakeRequestHash = web3.utils.soliditySha3('invalid');
+            utils.expectRevert(gatewayComposer.acceptStakeRequest(
+                invalidStakeRequestHash,
+                r,
+                s,
+                v,
+                hashLock,
+                { from: facilitator },
+            ),
+            'Should revert as stake request not found.',
+            'Stake request not found.');
+        });
+
+        it('Fails when BT.acceptStakeRequest() require fails.', async () => {
+            const {
+                gatewayComposer,
+                valueToken,
+                brandedToken,
+                owner,
+            } = await gatewayComposerUtils.setupGatewayComposer(
+                accountProvider,
+                false,
+            );
+
+            const {
+                stakeAmount,
+            } = await gatewayComposerUtils.setupGatewayComposerRequestStake(
+                valueToken,
+                gatewayComposer,
+                owner,
+            );
+
+            const {
+                gateway,
+                facilitator,
+            } = await gatewayComposerUtils.setupGatewayPass(
+                accountProvider,
+            );
+
+            const mintAmount = await brandedToken.convertToBrandedTokens(stakeAmount);
+            const beneficiary = accountProvider.get();
+            const gasPrice = 1;
+            const gasLimit = 1;
+            const nonce = 1;
+            const stakeRequestHash = await gatewayComposer.requestStake.call(
+                stakeAmount,
+                mintAmount,
+                gateway.address,
+                beneficiary,
+                gasPrice,
+                gasLimit,
+                nonce,
+                { from: owner },
+            );
+
+            const transactionResponse = await gatewayComposer.requestStake(
+                stakeAmount,
+                mintAmount,
+                gateway.address,
+                beneficiary,
+                gasPrice,
+                gasLimit,
+                nonce,
+                { from: owner },
+            );
+
+            assert.strictEqual(transactionResponse.receipt.status, true);
```

Comment:
Global comment: I don't believe this assert is needed, because this unit test does not need to concern itself with asserting that the EVM is operating correctly. Suggest removing.